### ISSUE TITLE
GHA artifacts not being published on HUD if multiple workflow runs exist for same workflow

### ISSUE THREAD

**Issue by jithunnair-amd** [pytorch/test-infra#5298]
If a commit has `mem_leak_check` and `rerun_disabled_tests` jobs running along with the regular `trunk` workflows jobs, we find that the HUD page for the commit doesn't list the artifacts for the regular `trunk` workflow. E.g. https://hud.pytorch.org/pytorch/pytorch/commit/cf77e7dd9770caf65e898ac2ee82045aa0408e30#rocm
The `linux-focal-rocm6.1-py3.8 / test (default, 1, 6, linux.rocm.gpu.2` jobs have a link for artifacts for the `mem_leak_check` and `rerun_disabled_tests` variants, but not for the regular one.

![image](https://github.com/pytorch/test-infra/assets/37884920/d0a15cba-66fe-4309-82f9-6d261b3bd99b)

However, from the corresponding Github Actions page for the regular job, https://github.com/pytorch/pytorch/actions/runs/9363246389/job/25774767756#step:19:51, it looks like the artifact was succesfully uploaded to https://gha-artifacts.s3.amazonaws.com/pytorch/pytorch/9363246389/1/artifact/test-jsons-test-default-1-6-linux.rocm.gpu.2_25774767756.zip

This is because there are two different workflow runs for the `rocm` workflow jobs in question above:
https://github.com/pytorch/pytorch/actions/runs/9363246389 (which runs the "regular" jobs)
https://github.com/pytorch/pytorch/actions/runs/9363869290 (which runs the "mem_leak_check" and "rerun_disabled_tests" jobs)

**Comment by clee2000** [comment]
For whomever ends up taking it up, its because we show multiple workflow runs in a box but only query for the artifacts of one workflow id

**Comment by huydhn** [comment]
AI: Need to double check if this is a ROCM-thing or if this is a wide spread issue with other runners too

**Comment by jithunnair-amd** [comment]
@clee2000 @huydhn Exorcising this ghost: this is still an issue and it's an issue for CUDA too: https://hud.pytorch.org/pytorch/pytorch/commit/75c8b7d9725af59d8348379a4165d1252c4ac208#pull

![Image](https://github.com/user-attachments/assets/c3868783-5d89-4c61-80d7-0f8427bc7a5e)

![Image](https://github.com/user-attachments/assets/6c006032-afbb-4ceb-a577-221e7a81bfa4)

![Image](https://github.com/user-attachments/assets/e686993d-39cb-4a4d-b284-2b8f5aebd567)

**Comment by jithunnair-amd** [comment]
> For whomever ends up taking it up, its because we show multiple workflow runs in a box but only query for the artifacts of one workflow id

Yes, this is the reason. There seems to be an implicit assumption somewhere in the HUD artifact code that there would only be one workflow run id for a given workflow and a given SHA.

**Comment by huydhn** [comment]
I'm bringing this out of cold storage so we know we still need to fix it (cc @clee2000 @ZainRizvi )
